pertpy/tools/_metadata.py

```diff
@@ -0,0 +1,118 @@
+            copy: Determines whether a copy of the `adata` is returned. (default: False)
+
+        Returns:
+            Returns an AnnData object with cell line annotation.
+        """
+        if copy:
+            adata = adata.copy()
+        if cell_line_type in self.cell_line_meta.columns:
+            """If the specified cell line type can be found in the DepMap database,
+            we can compare these keys and fetch the corresponding metadata."""
+            if cell_line_information is None:
+                """If no cell_line_information is specified, all metadata is fetched by default.
+                Sometimes there is already different cell line information in the `adata`.
+                In order to avoid redundant information,
+                we will remove the duplicate information from metadata after merging."""
+                adata.obs = (
+                    adata.obs.reset_index()
+                    .merge(
+                        self.cell_line_meta,
+                        left_on=cell_line_column,
+                        right_on=cell_line_type,
+                        how="left",
+                        suffixes=("", "_fromMeta"),
+                    )
+                    .filter(regex="^(?!.*_fromMeta)")
+                    .set_index(adata.obs.index.names)
+                )
+                """ If cell_line_column and cell_line_type have different names,
+                there will be a column for each of them after merging,
+                which is redundant as they refer to the same information.
+                We will move the cell_line_type column."""
+                if cell_line_column != cell_line_type:
+                    del adata.obs[cell_line_type]
+
+            elif set(cell_line_information).issubset(set(self.cell_line_meta.columns)):
+                """If cell_line_information is specified and can be found in the DepMap database,
+                We will subset the original metadata dataframe correspondingly and add them to the `adata`.
+                Again, redundant information will be removed."""
+                if cell_line_type not in cell_line_information:
+                    cell_line_information.append(cell_line_type)
+                cell_line_meta_part = self.cell_line_meta[cell_line_information]
+                adata.obs = (
+                    adata.obs.reset_index()
+                    .merge(
+                        cell_line_meta_part,
+                        left_on=cell_line_column,
+                        right_on=cell_line_type,
+                        how="left",
+                        suffixes=("", "_fromMeta"),
+                    )
+                    .filter(regex="^(?!.*_fromMeta)")
+                    .set_index(adata.obs.index.names)
+                )
+                """ Again, redundant information will be removed."""
+                if cell_line_column != cell_line_type:
```

Comment:
Please correct me if i'm wrong, but I don't see what happens if some `cell_line_identifier`s cannot be found in the `identifier_type` column. If it's in here, could you please also include a description of how these are ignored in the docstring?